crates/aide-macros/src/lib.rs

```diff
@@ -166,3 +166,25 @@ pub fn derive_operation_io(ts: TokenStream) -> TokenStream {
 
     ts.into()
 }
+
+#[cfg(feature = "axum-extra-typed-routing")]
+/// Example usage:
+/// ```ignore
+/// #[aide::axum_typed_path]
+/// #[typed_path("/foo/bar")]
+/// struct FooBar;
+/// ```
+#[proc_macro_attribute] // functions tagged with `#[proc_macro_attribute]` must currently reside in the root of the crate
+pub fn axum_typed_path(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    use syn::{Attribute, ItemStruct};
+    let mut input = parse_macro_input!(item as ItemStruct);
+    let mut attrs: Vec<Attribute> = syn::parse_quote! {
+        #[derive(::axum_extra::routing::TypedPath, ::aide_macros::OperationIo, ::schemars::JsonSchema, ::serde::Deserialize)]         
+        #[aide(input_with = "axum::extract::Path<Self>")]
+    };
+    let mut new_attrs = Vec::with_capacity(input.attrs.len() + 1);
+    new_attrs.append(&mut attrs);
+    new_attrs.append(&mut input.attrs);
+    input.attrs = new_attrs;
+    TokenStream::from(quote!(#input))
```

Comment:
You don't even need to parse the input if all you want to do is apply some derives:

```suggestion
    let input = proc_macro2::TokenStream::from(input);
    quote! {
        #[derive(
            ::axum_extra::routing::TypedPath,
            ::aide_macros::OperationIo,
            ::schemars::JsonSchema,
            ::serde::Deserialize,
        )]
        #[aide(input_with = "axum::extract::Path<Self>")]
        #input
    }.into()
```